src/core/tsi/ssl_transport_security.cc

```diff
@@ -1952,6 +1989,22 @@ tsi_result tsi_create_ssl_client_handshaker_factory_with_options(
     SSL_CTX_set_session_cache_mode(ssl_context, SSL_SESS_CACHE_CLIENT);
   }
 
+  if (options->key_logger != nullptr) {
+    // Unref is manually called on factory destruction
+    impl->key_logger =
+        reinterpret_cast<tsi::TlsKeyLogger*>(options->key_logger)->Ref();
+    if (options->session_cache == nullptr) {
+      // Need to set factory at g_ssl_ctx_ex_factory_index
+      SSL_CTX_set_ex_data(ssl_context, g_ssl_ctx_ex_factory_index, impl);
```

Comment:
Can we restructure the code to 
```
if (options->session_cache != nullptr) {
    impl->session_cache = ...;
    SSL_CTX_sess_set_new_cb(...);
    SSL_CTX_set_session_cache_mode(...);
}
if (options->key_logger != nullptr) {
    impl->key_logger = ...;
    SSL_CTX_set_keylog_callback(...);
}
if (options->session_cache != nullptr || options->key_logger != nullptr) {
   SSL_CTX_set_ex_data(...);
}
```
We can probably also consider removing the last condition enclosing `SSL_CTX_set_ex_data`, since I think the cost is minimal(setting a pointer).

Reply:
Done.